### DISCUSSION TITLE
API中更新用户接口put请求（/api/v1/user/{id}/），报错

### DISCUSSION THREAD

**Discussion by orangeAndoranges** [hhyo/Archery#2596] [Q&A]
### 重现步骤

1.  使用接口 /api/v1/user/{id}/
2. PUT请求
3. 已将路径中{id}替换为自己在archery系统中的用户id
4. 请求体中设置了信息
但是接口响应结果一直是500

### 预期外的结果

![image](https://github.com/hhyo/Archery/assets/45198957/71d64925-6489-4055-92fa-e8d53acd097e)

### 版本

v1.8.5

### 部署方式

Docker

**Comment by nick2wang** [comment]
请求报文发出来看下

**Reply by orangeAndoranges** [reply]
我是想修改用户拥有的用户组权限，理想传参body是下面这个(试了不行，响应500)：
{
  "groups": [
    1,7
  ]
}

然后我把很多参数都写上，如下（这样也是响应500）：
{
  "is_superuser": false,
  "username": "XXX",
  "email": "XXX.com",
  "is_staff": false,
  "is_active": true,
  "display": "汉字名称",
  "date_joined":"2023-08-21 17:37:06",
  "groups": [
    1,7
  ],
  "user_permissions": [],
  "resource_group": []
}

**Comment by nick2wang** [comment]
应该是这个问题#1776 1.9.0版本已经修复，建议更新版本

**Reply by orangeAndoranges** [reply]
但是我也使用了官网提供的demo那个地址中的swagger上面的接口，官网demo上面是v1.10版本，也是报错500。官网demo接口地址：https://demo.archerydms.com/api/swagger/#/

**Reply by orangeAndoranges** [reply]
好像修复的没有彻底。测试了v1.10.0版本的API更新用户接口，只能增加用户组权限，不能减少用户组权限。
比如增加用户组权限body传参：
{
  "groups": [
    1,2,3
  ]
}
然后如果我想减少某个用户的用户组权限传参body：
{
  "groups": [
    1,2
  ]
}
这样接口也是响应500

**Reply by nick2wang** [reply]
demo环境我不清楚，最新代码我本地测试是没有问题的，你可以拉下master代码验证下

**Reply by orangeAndoranges** [reply]
好的，我试试。感谢
